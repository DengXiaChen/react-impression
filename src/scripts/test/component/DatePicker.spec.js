import React from 'react';
import { mount, shallow } from 'enzyme';
import { Input } from '../../components/impression';

const setup = () => {
    const props = {
        defaultValue: '2017-7-1',
        showToday: true,
        todayText: 'today',
        firstDayOfWeek: 1,
        minDate: '2017-7-1',
        maxDate: '2018-7-1',
        onSelect: jest.fn(),
        onChange: jest.fn(),
        disabled: false,
    };

    const wrapper = mount(
        <Input type="date" {...props} />
    );

    const wrapper1 = shallow(
        <Input type="date" {...props} />
    );

    return {
        props,
        wrapper,
        wrapper1,
    };
};

describe('CheckboxGroup', () => {
    const { wrapper, wrapper1, props } = setup();

    it('render', () => {
        expect(wrapper.find('.input').exists()).toBe(true);
        // 激活datePicker
        wrapper.find('input').simulate('click');
        expect(wrapper.find('.datepicker').exists()).toBe(true);
        // 自定义today
        expect(wrapper.find('.datepicker-footer a').text()).toBe('today');
        // 从周一开始
        expect(wrapper.find('.datepicker-weekgroup-item').at(0).text()).toBe('一');
        // 存在日期限制
        expect(wrapper.find('.datepicker-daygroup-item-text.disable').exists()).toBe(true);
    });

    it('click', () => {
        wrapper.find('.datepicker-daygroup-item-text').findWhere(n => n.text() === '31').simulate('click');
        expect(props.onChange).toHaveBeenCalledWith('2017-07-31');
    });

    it('ui', () => {
        wrapper.find('input').simulate('click');
        expect(wrapper1).toMatchSnapshot();
    });
});
