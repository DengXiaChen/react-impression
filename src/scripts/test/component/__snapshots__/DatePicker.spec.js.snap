// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxGroup ui 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="input"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
>
    <input
        className="form-control input-field input-field-addon"
        defaultValue="2017-7-1"
        disabled={false}
        onClick={[Function]}
        placeholder={undefined}
        readOnly={true}
        style={undefined}
        type="text"
        value={undefined}
    />
    <i
        className="fa fa-calendar input-addon"
        onClick={[Function]}
    />
</div>,
  "nodes": Array [
    <div
      className="input"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
>
      <input
            className="form-control input-field input-field-addon"
            defaultValue="2017-7-1"
            disabled={false}
            onClick={[Function]}
            placeholder={undefined}
            readOnly={true}
            style={undefined}
            type="text"
            value={undefined}
      />
      <i
            className="fa fa-calendar input-addon"
            onClick={[Function]}
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <Input
        clearable={true}
        defaultValue="2017-7-1"
        disabled={false}
        firstDayOfWeek={1}
        maxDate="2018-7-1"
        minDate="2017-7-1"
        onChange={[Function]}
        onSelect={[Function]}
        showToday={true}
        todayText="today"
        type="date"
/>,
      "_debugID": 6,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Input {
        "_reactInternalInstance": [Circular],
        "clearInputHandle": [Function],
        "context": Object {},
        "hideClearHandle": [Function],
        "hideOptionsHandle": [Function],
        "props": Object {
          "clearable": true,
          "defaultValue": "2017-7-1",
          "disabled": false,
          "firstDayOfWeek": 1,
          "maxDate": "2018-7-1",
          "minDate": "2017-7-1",
          "onChange": [Function],
          "onSelect": [Function],
          "showToday": true,
          "todayText": "today",
          "type": "date",
        },
        "refs": Object {},
        "selectOptionsHandle": [Function],
        "showClearHandle": [Function],
        "showOptionHandle": [Function],
        "state": Object {
          "showClear": false,
          "showOption": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="input"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
>
          <input
                    className="form-control input-field input-field-addon"
                    defaultValue="2017-7-1"
                    disabled={false}
                    onClick={[Function]}
                    placeholder={undefined}
                    readOnly={true}
                    style={undefined}
                    type="text"
                    value={undefined}
          />
          <i
                    className="fa fa-calendar input-addon"
                    onClick={[Function]}
          />
</div>,
        "_debugID": 7,
        "_renderedOutput": <div
          className="input"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
>
          <input
                    className="form-control input-field input-field-addon"
                    defaultValue="2017-7-1"
                    disabled={false}
                    onClick={[Function]}
                    placeholder={undefined}
                    readOnly={true}
                    style={undefined}
                    type="text"
                    value={undefined}
          />
          <i
                    className="fa fa-calendar input-addon"
                    onClick={[Function]}
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Input
    clearable={true}
    defaultValue="2017-7-1"
    disabled={false}
    firstDayOfWeek={1}
    maxDate="2018-7-1"
    minDate="2017-7-1"
    onChange={[Function]}
    onSelect={[Function]}
    showToday={true}
    todayText="today"
    type="date"
/>,
}
`;
